### PR TITLE
Reorganize Conversation section

### PR DESCRIPTION
**Pull request by alavers**
Split it into three new top level sections:
- Message
- Message Types
- Message Actions

Before (Post Message was friggin huge):
before

After:
after

### CODE REVIEW

File: slate/source/includes/rest/core/_message.md [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+<api>`POST /v1/apps/{appId}/appusers/{smoochId|userId}/images`</api>
+
+Upload an image and post it to the conversation. Images are uploaded using the `multipart/form-data` content type. Similar to the `/messages` endpoint, a `role` parameter must be specified. The `/images` endpoint accepts the same parameters as `/messages` but they are sent as form parameters as opposed to being encoded in JSON bodies. The uploaded image will render as part of the message thread in all supported app maker channels (email, Slack, HipChat, Zendesk, Helpscout).
+
+| **Form Parameters**          |                            |
+|------------------------------|----------------------------|
+| **source**<br/><span class='req'>required</span>    | The image data.            |
+| **role**<br/><span class='req'>required</span>      | The role of the individual posting the message. Can be either `appUser` or `appMaker`. |
+
+## Typing Activity
```
Comment (mspensieri) [inline slate/source/includes/rest/core/_message.md]: It's a bit odd that `Typing Activity` and `Reset Unread Count` fall under the `Message` section now, `Conversation` was more descriptive of these endpoints /shrug

Comment (alavers) [reply slate/source/includes/rest/core/_message.md]: The conversation concept may evolve into a first class citizen the future and these two sections would naturally live there. Do you think I should I just split these two into a separate Conversation section now?

Comment (mspensieri) [reply slate/source/includes/rest/core/_message.md]: I think it would be a bit clearer, do you agree?

Comment (alavers) [reply slate/source/includes/rest/core/_message.md]: 👍 Done

**Pull request merged**